source/dagger/gameplay/team_game/golem.cpp

```diff
@@ -0,0 +1,180 @@
+#include "golem.h"
+
+#include "mage.h"
+#include "range_of_attack.h"
+
+#include "core/engine.h"
+#include "core/game/transforms.h"
+#include "core/graphics/animation.h"
+#include "core/graphics/sprite.h"
+
+#include "gameplay/common/simple_collisions.h"
+
+#include <math.h>
+#include <list>
+
+using namespace ancient_defenders;
+
+
+void ancient_defenders::GolemBehaviorSystem::SpinUp()
+{
+	Engine::Dispatcher().sink<NextFrame>().connect<&GolemBehaviorSystem::OnEndOfFrame>(this);
+}
+
+void ancient_defenders::GolemBehaviorSystem::WindDown()
+{
+	Engine::Dispatcher().sink<NextFrame>().disconnect<&GolemBehaviorSystem::OnEndOfFrame>(this);
+}
+
+void ancient_defenders::GolemBehaviorSystem::Run()
+{
+	Engine::Registry().view<GolemStats, Sprite, Transform, Animator, RangeOfAttack>().each(
+		[](GolemStats& golem_, Sprite& sprite_, Transform& transform_, Animator& animation_, RangeOfAttack& range_)
+		{
+			if (range_.targetFound) golem_.currentAction = EAction::Attacking;
+
+			if (golem_.currentAction == EAction::Idling) {
+				AnimatorPlay(animation_, "ancient_defenders:golem:IDLE");
+			}
+			else if (golem_.currentAction == EAction::Moving) {
+				auto nextPosition = golem_.postition - 1;
+
+				auto coords = WalkingPath::path.back();
```

Comment:
You can use auto coords = WalkingPath::path[nextPosition]; since the path will be switched from list to vector.

Reply:
And then you don't need the loop for finding the element.